src/components/InstructionsErrorBoundary.jsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import markdown from '../util/markdown';
+
+export default class InstructionsErrorBoundary extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {useHighlighting: true};
+  }
+
+  unstable_handleError() { // eslint-disable-line camelcase
+    /*
+    Highlight.js throws an exception when fed a language it doesn't have
+    configured. None of the React wrappers we're using catch that error,
+    and since it gets run in the child React components created from markdown
+    code sections, these exceptions occur outside the current callstack.
+    `unstable_handleError` is a kludge for catching exceptions raised in child
+    renders introduced in React 15 and to be replaced by `componentDidCatch` in
+    React 16.
+    <InstructionsErrorBoundary> exists because `unstable_handleError` only
+    wraps the *initial* mounting of a child -- we need an "error boundary"
+    component that comes into being in the same mounting phase that
+    Highlight.js gets run.
+    In theory this component can be removed and replaced with a
+    `componentDidCatch` method on the parent when we upgrade to React 16.
+     */
+    // eslint-disable-next-line react/no-set-state
+    this.setState({useHighlighting: false});
```

Comment:
So, not a hill I want to die on, but having given it more thought I think there is a strong argument for pushing this to Redux, not in the form of “there was an error here” but rather in the form of “the instructions are known to break syntax highlighting”. So, a property of the instructions rather than the UI. WDYT?

Reply:
Well, it's not like putting it in Redux hurts at all. I don't really see a benefit, but I can do it for the sake of consistency within the project. Where in the redux state structure would you put it?

Reply:
Yeah I guess I see two appeals:

* Consistency, as you mention—this would be the only `setState` in the code base
* Also, I am a partisan of the spirit behind “everything in Redux”, particularly the claim that the UI state *does* derive entirely from an underlying state of the world, and that the React layer is most legible when its job is just translating that state of the world into a DOM. Obviously the `ui` tree in the Redux store does betray the notion that sometimes UI state is just UI state, and libraries like https://github.com/tonyhb/redux-ui and others (which I’m not inherently against) make that even more explicit. But in this case what we’re talking about is actually a fact about the instructions, namely that they can’t be rendered with the syntax highlighting code we have in place. And I do think there is value to making that explicit in the “one true state of the world”.

As for a specific proposal, I think I’d put it in the project, basically turning `instructions` into a record with properties along the lines of `content` and `isKnownToBreakSyntaxHighlighting`.

All that said, I’m definitely open to being told I’m wrong/being too much of a purist/etc.!